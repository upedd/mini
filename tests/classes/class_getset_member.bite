native print;

class Foo {
    _a;
    get a {
        print("foo.a");
        return _a;
    }

    set a {
        print("=foo.a");
        return _a;
    }

    pow() {
        a *= 2;
    }

    set_a(a) {
        this.a = a * 10;
    }
}


let instance = Foo();
instance.set_a(1);
instance.pow();
instance.pow();
print(instance.a);

let instance2 = Foo();
instance2.set_a(3);
instance2.pow();
print(instance2.a);

print(instance.a);