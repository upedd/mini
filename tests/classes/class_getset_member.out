=foo.a
foo.a
=foo.a
foo.a
=foo.a
foo.a
40
=foo.a
foo.a
=foo.a
foo.a
60
foo.a
40